### ISSUE TITLE
Text Classification in Hosted Inference API with Multiple Inputs

### ISSUE THREAD

**Issue by merveenoyan** [huggingface/huggingface_hub#567]
Some of the text classification tasks take two inputs just like similarity models, but since the problem is text classification, widget takes only one text, which confuses users on how to pass their text to model in hosted inference API. See [this question in the forum](https://discuss.huggingface.co/t/two-texts-inputs-for-text-classification-in-inference-api/12805). 
Couple of other example models: [https://huggingface.co/cross-encoder/qnli-electra-base?text=I+like+you.+I+love+you](Cross Encoder QNLI) (even though these models are based on similarity, they usually return entailment/not entailment)
[Cross Encoder MSMARCO Passage Ranking](https://huggingface.co/cross-encoder/ms-marco-MiniLM-L-12-v2) 

Solution is letting user input as many text as they want (like similarity widget) yet keeping the class labels without having an additional pipeline.

**Comment by Narsil** [comment]
pipeline-wise, it seems that this task is more like pair-wise classification, NOT pure classification.

zero-shot uses under the hood entailment to do the job, but it does not seems like the pipeline will be reusable (since 1 sentence, comma-separated labels -> classification outputs) is not really what is desired here (couple of sentences -> Entailment/not entailment/neutral classification).

Couple of notes/questions:

- How many models require this to be showcased properly ? (Or how popular are they too?, just to gauge a sense of variability in what would be a nice showcase for those models).
- If we send more than two sentences, what is the expected output ?
- Maybe we need to keep in mind backward compatiliby vs Zero-shot which I think is the default pipeline for these models (mnli variants at least).
- There is a way to make zero-shot work on exactly the same input with `hypothesis` parameter and by enabling `multi_label`. (Definitely not ideal, just food for though)

**Comment by merveenoyan** [comment]
Hello @Narsil 🤗
Yes, it's just there's this family of tasks in text classification called GLUE and it's way too general (reason why we needed zero shot classification, I guess). For [QNLI](https://huggingface.co/models?pipeline_tag=text-classification&sort=downloads&search=qnli) ones we can simply change the name of the input texts in the widget imo, it only takes a question and a context. There's another task called [QQP](https://huggingface.co/models?pipeline_tag=text-classification&sort=downloads&search=qqp) that assesses if two questions are similar, takes two separate inputs. Another one is [MRPC](https://huggingface.co/models?pipeline_tag=text-classification&sort=downloads&search=mrpc) which again takes two texts.
TLDR; some of the GLUE tasks take one input (which is covered), some take two inputs (not covered) and others are covered under zero shot. The similarity based ones are not text classification technically (the MSMARCO I put above) so it's okay if we don't cover it for now imo, I think it's because the way these similarity models work (they're not actually classification models). 
The ones that take two inputs can be solved with same pipeline but we should change the name of the input according to task itself, somehow.

**Comment by Narsil** [comment]
I see ! 

Right now I don't see anything around having a new pipeline for processing texts two at a time in a classification manner (`text-classification` cannot be realistically extended, since `pipe(["text1", "text2"])` is already defined and just means classify two texts). I am also not in favor of mixing argument types all the time.

`sentence-similarity` fits the bill perfectly and already exists as a widget, @osanseviero wdyt, should we add it to `transformers` ? IMHO it seems like the best course of action.

**Comment by osanseviero** [comment]
> sentence-similarity fits the bill perfectly and already exists as a widget, @osanseviero wdyt, should we add it to transformers ? IMHO it seems like the best course of action.

Just as a note, I see many models are using Crossencoder which is a `sentence-transformers` class, so maybe we should also consider moving some models to `sentence-transformers`. E.g.
* https://huggingface.co/cross-encoder/ms-marco-MiniLM-L-12-v2
* https://huggingface.co/cross-encoder/qnli-electra-base
* etc

Anyways, I think `sentence-similarity` name is a bit strange for this case, no? I think the pipeline should have a different name although we can reuse the widget. My main question is if the inference of all these models that are currently not supported consistent. That is, if all models expect the two inputs in the same way.

**Comment by merveenoyan** [comment]
@osanseviero QQP and MRPC models answer if one sentence is a paraphrase of another so they're irrelevant with `sentence-similarity` yet take two inputs (they're in GLUE).

**Comment by merveenoyan** [comment]
It was raised by another user [in this forum question](https://discuss.huggingface.co/t/pipeline-text-classification-with-two-sequences-for-each-example/15083). Two sentences is just fine but with more sentences it might be problematic.
